Core/Src/main.c

```diff
@@ -97,6 +120,30 @@ int main(void)
   {
     /* USER CODE END WHILE */
 
+	  //chip select is low to start transmission - data is read from ADC using SPI
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  //timeout timing to 100ms - seemed like a reasonable value
+	  HAL_SPI_TransmitRecieve(&hspi1, transmit_Data, recieve_data, sizeof(transmit_data), 100);
```

Comment:
100ms is reasonable! But make it a macro too since 100 is a magic number! Also use the macro for the size that we talked about above :)